src/Monaco/mylang.js

```diff
@@ -0,0 +1,579 @@
+        mylangTables = mylangTables.map((tables) => tables[1]); // Cuts off the name.
+        mylangTables = [].concat(...mylangTables);  // Concatenate all the tables into one array.
+
+        console.log("Original Tables before transformation")
+        console.log(mylangTables)
+
+        // Transforms the mylangTables into a Monaco Element
+        mylangTables = mylangTables.map(t => {
+          return {
+            label: t.label,
+            kind: monaco.languages.CompletionItemKind.Field,
+            insertText: t.label,
+            detail: t.value + ""
+          };
+        })
+      })
+      .catch(error => console.error(error));
+  }
+
+  /**
+   * Mock method, will be replaced by eonum
+   * @param {*} filePath file path to the JSON file
+   * @returns a Promise with the data and a status of the asyncronous function
+   * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise
+   */
+  async function loadVariablesFromJSON(filePath){
+    return fetch(filePath)
+      .then(response => response.json())  // Turn the JSON into a JS object.
+      .then(varData => {
+        // mylangVariables = varData;
+        mylangVariables = Object.entries(varData);  // Transforms everything into an Object.
+        mylangVariables = mylangVariables.map(variables => variables[1]);  // Cuts off the name.
+
+        console.log("Original Variables before transformation")
+        console.log(mylangVariables)
+
+        // Transforms the mylangVariables into a Monaco Element
+        mylangVariables = mylangVariables.map(v => {
+          // Set the correct documentation
+          let options = ""
+          if(!v.is_enum){
+            options = "\nAlles vom Typ " + v.variable_type
+          }
+          else{
+            //options = v.values
+            let count = 1;
+            for(opt of v.values_de){
+              options += "\n" + count + ") " + opt;
+              count += 1;
+            }
+          }
+
+          return {
+            label: v.field_name,
+            kind: monaco.languages.CompletionItemKind.Variable,
+            insertText: v.field_name,
+            detail: v.name_de,
+            documentation: "Mögliche Werte : " + options
+          };
+        })
+      })
+      .catch(error => console.error(error));
+  }
+
+  /**
+   * A getter for all the lists
+   * @returns an Array with all the Lists concatenated
+   */
+  function getAllLists(){
+    return mylangKeywords
+    .concat(mylangFunctions)
+    .concat(mylangVariables)
+    .concat(mylangTables);
+  }
+
+  // Links to the JSON files
+  const varFilePath = '../../variables.json';
+  const tabFilePath = '../../tables.json';
+
+  let mylangVariables;
+  let mylangTables;
+
+  /** @todo There is a bug where after entering a keyword the editor doesn't work properly */
+  /** @todo Are these really necessary / needed in medcoedelogic? */

```

Comment:
Not all of them. You find the keywords in the file we provided (the presentation about medcodelogic language). There you fing The slide "The grammar / Expressions" from which you can take the keywords (f.e. or, and, ...)

Reply:
So only [or, and] are needed?
<img width="1047" alt="image" src="https://github.com/eonum/mcl-editor/assets/107117424/99894c03-4e44-42de-b4ed-bd0ece9ccbf8">

Reply:
yes, i thinkt that's it.